### PR TITLE
Ben/dev 80 - Add Profile/Settings Page

### PR DESCRIPTION
**Pull request by bLopata**
Also takes care of Dev-294 & Dev-457
`/settings` route for adjusting profile values for name/image/email/password, email support, adding discord integration using `supabase.auth.linkIdentity`

General overview:
https://app.screencast.com/V9FPfop8clwbu

Discord integration:
https://app.screencast.com/VrHCVDCh16FXs

### CODE REVIEW

File: www/components/sidebar.tsx [hunk 1]
```diff
@@ -94,17 +96,32 @@ export default function Sidebar({
...
-      className={`fixed lg:static z-20 inset-0 flex-none h-full w-full lg:absolute lg:h-auto lg:overflow-visible lg:pt-0 lg:w-60 xl:w-72 lg:block lg:shadow-lg border-r border-gray-300 dark:border-gray-700 ${isSidebarOpen ? '' : 'hidden'
-        }`}
+      className={`fixed z-20 inset-0 flex-none h-full w-full lg:absolute lg:h-auto lg:overflow-visible lg:pt-0 lg:w-60 xl:w-72 lg:block lg:shadow-lg border-r border-gray-300 dark:border-gray-700 ${
+        isSidebarOpen ? '' : 'hidden'
+      }`}
     >
       <div
-        className={`h-full scrollbar-trigger overflow-hidden bg-white dark:bg-gray-950 dark:text-white sm:w-3/5 w-4/5 lg:w-full flex flex-col ${isSidebarOpen ? 'fixed lg:static' : 'sticky'
-          } top-0 left-0`}
+        className={`h-full scrollbar-trigger overflow-hidden bg-white dark:bg-gray-950 dark:text-white sm:w-3/5 w-4/5 lg:w-full flex flex-col ${
```
Comment (VVoruganti) [inline www/components/sidebar.tsx:122]: image

Text in sidebar is white in dark mode, makes it hard to read, should stay black the entire time

File: www/app/page.tsx [hunk 2]
```diff
@@ -289,11 +279,7 @@ export default function Home() {
   }
 
   return (
-    <main
-      className={`flex h-[100dvh] w-screen flex-col pb-[env(keyboard-inset-height)] text-sm lg:text-base overflow-hidden relative ${
-        isDarkMode ? 'dark' : ''
-      }`}
-    >
+    <main className="flex w-screen flex-col pb-[env(keyboard-inset-height)] text-sm lg:text-base overflow-hidden relative h-full">
       <Sidebar
```
Comment (VVoruganti) [inline www/app/page.tsx:283]: Can't open the left sidebar in mobile
image

**Comment by VVoruganti** [comment]
image
Should have some kind of visual distinction between top bar and the start of the sidebar

**Review by VVoruganti — CHANGES_REQUESTED** [review]
Going to try to consolidate my feedback in this comment, 

This is the settings sidebar and there should some visual indication of which page is selected, along with a visual indication of a hover i. e. `transform background color`

image

Horizontal Bar should be above the Email Label to associate it with that one

image

It is failing to `pnpm run build`

image


- [ ] Also make sure you run lint and pretty

File: www/package.json [hunk 3]
```diff
@@ -11,12 +11,20 @@
...
+    "@radix-ui/react-slot": "^1.1.0",
+    "@radix-ui/react-tabs": "^1.1.1",
     "@sentry/nextjs": "^8.34.0",
     "@stripe/stripe-js": "^4.8.0",
     "@supabase/ssr": "^0.5.1",
     "@supabase/supabase-js": "^2.45.4",
+    "class-variance-authority": "^0.7.0",
+    "clsx": "^2.1.1",
     "katex": "^0.16.11",
+    "lucide-react": "^0.452.0",
```
Comment (VVoruganti) [inline www/package.json:25]: We shouldn't be using `lucide-react` and `react-icons` basically let's try to only use one icon library

**Comment by bLopata** [comment]
@VVoruganti addressed all your comments
https://app.screencast.com/EEwkjY4NPydpM

**Comment by VVoruganti** [comment]
Do you have a list of edge cases that were tested with the `linkIdentity` feature and the behavior?

**Review by VVoruganti — CHANGES_REQUESTED** [review]
image

Buttons on Auth Pages don't have proper dark mode handling

Seeing weird behavior when testing manual linking. Steps to reproduce

1. Have one account that is authenticated with an email
2. Have another account that is authenticated via discord with a different email 

From the first account try to connect your discord identity. When you connect via the oauth flow, the screen will return to the settings page with an `?error` in the url address, however the user will not be authenticated anymore and no errors will print or register in the logic handler. 

Since the user's authentication session is invalidated they should be redirected to the /auth screen. May need to manually add a `redirect` call in the `SettingsPage`

I'm seeing the error in the url, but I don't know why it is invalidating the auth session. 

---
Another edge case is if I have a user with an account that is connected via Discord then I unlink my discord account I then have no password associated with the account. 

Going through the password recovery flow returns an error.

**Review by VVoruganti — APPROVED** [review]

**Pull request merged**